core/src/main/java/com/linecorp/armeria/common/util/EventLoopThreadFactory.java

```diff
@@ -19,48 +19,55 @@
 import static java.util.Objects.requireNonNull;
 
 import java.util.concurrent.ThreadFactory;
+import java.util.function.Function;
 
 import javax.annotation.Nullable;
 
-import io.netty.util.concurrent.DefaultThreadFactory;
-
 /**
  * {@link ThreadFactory} that creates event loop threads.
  *
  * @see EventLoopGroups
+ *
+ * @deprecated Use {@link ThreadFactories#newEventLoopThreadFactory(String, boolean)} or
+ * {@link ThreadFactories#builder(String)}. Note that setting
+ * {@link ThreadFactoryBuilder#eventLoop(boolean)} true is required to create event loop thread. e.g.
+ * {@code ThreadFactories.builder("treadPrefix").eventLoop(true).build()}.
  */
-public final class EventLoopThreadFactory implements ThreadFactory {
-
-    // Note that we did not extend DefaultThreadFactory directly to hide it from the class hierarchy.
-    private final ThreadFactory delegate;
-
+@Deprecated
+public final class EventLoopThreadFactory extends AbstractThreadFactory {
     /**
      * Creates a new factory that creates a non-daemon and normal-priority thread.
      *
      * @param threadNamePrefix the prefix of the names of the threads created by this factory.
+     *
      */
     public EventLoopThreadFactory(String threadNamePrefix) {
-        this(new EventLoopThreadFactoryImpl(requireNonNull(threadNamePrefix, "threadNamePrefix")));
+        super(requireNonNull(threadNamePrefix, "threadNamePrefix"), false, Thread.NORM_PRIORITY,
```

Comment:
I think we could remove `requireNonNull(threadNamePrefix, "threadNamePrefix")` on this method and other constructors because super method always checks null.

Reply:
Removed all redundant `requireNonNull`.